### PR TITLE
RNN.call should get initial state from full input spec

### PR DESCRIPTION
**Pull request by yanboliang**
### Summary
This PR fix a critical bug in ```RNN``` which reported at #9449 and #10830 .

In ```RNN.call```, if ```initial_state``` is a tensor that was returned by a Keras layer, we should get ```initial_state``` from full input spec(including training data, state and constants) which was generated at ```RNN.__call__```, as it could be copied to multiple GPUs. Otherwise, it would use the original ```initial_states``` which is not be sliced according the number of GPUs.
BTW, I have also check ```CuDNNRNN```, it use the correct way, so we don't need to modify it.

I run the following test code in a machine with 2 GPUs, it works well after this fix.
```
import numpy as np
import keras
from keras import layers as L
from keras.models import Sequential, Model
from keras.utils.multi_gpu_utils import multi_gpu_model

x = L.Input((4,3))
init_state = L.Input((3,))
y = L.SimpleRNN(3,return_sequences=True)(x,initial_state=init_state)
_x = [np.random.randn(2,4,3),np.random.randn(2,3)]
_y = np.random.randn(2,4,3)
m = Model([x,init_state],y)
m2 = multi_gpu_model(m,2)
m2.compile(loss='mean_squared_error',optimizer='adam')
m2.train_on_batch(_x,_y)
```

### Related Issues
#9449
#10830
 
### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by mikejhuang** [comment]
I updated with the bugfix but I still get the InvalidArgumentError with multi_gpu_model enabled.

**Comment by JunHyungYu** [comment]
hi @mikejhuang i have same issue with you in convLSTM....

how did you solve your problem ?? 
  if you know about it , please tell me ... i can't train my seq2seq convLSTM model with initial_state...
somebody help me~~

**Comment by mikejhuang** [comment]
@JunHyungYu 
I used this bugfix
https://github.com/pascalxia/keras/commit/6750e1e51fb106d6595f212bd7a1fd1e95e0e91e

To get it to work, you input a list containing the initial states ([input, init_state_h, init_state_c])

**Comment by JunHyungYu** [comment]
@mikejhuang 
if i use convlstm()([input, init_state_h, init_state_c])   , i can make multi gpu model..

but when i train this multi gpu model, i have same error ..

can you train your convlstm with initial_state using multi gpu model????

**Comment by mikejhuang** [comment]
Nope, I haven't worked it out. Let me know if you find out.

**Comment by JunHyungYu** [comment]
@mikejhuang Oh my god.... thank you..

**Comment by dzhv** [comment]
@mikejhuang, @JunHyungYu  have any of you managed to find a workaround for the issue? I'm struggling with it atm :(

**Comment by maym2104** [comment]
Same issue with ConvLSTM2D w/ latest of Keras (2.2.4) and TensorFlow 1.13

**Comment by maym2104** [comment]
It would seem that instantiating the model on a CPU device (as we should) is a (temp) fix to the problem (using `with tf.device('/cpu:0'):`, see the first example of multi_gpu_model on https://keras.io/utils/). I was using cpu_relocation=True, but it did not do the trick.

**Comment by maym2104** [comment]
It only happens when I use this combination:
- ConvLSTM
- w/ recurrent dropout and dropout
- in a multi-gpu model.

It is working when I remove the dropout on this layer, but it results in a less efficient learning :/
